src/app.ts

```diff
@@ -183,9 +187,47 @@ export async function newApp(
   registerAPIRoutes(app, APIBasePath, allowNotifyIPSourceRange);
   registerPagoPARoutes(app, PagoPABasePath, allowPagoPAIPSourceRange);
 
+  const timer = idpMetadataUpdater(app);
+  app.on("server:stop", () => {
+    clearInterval(timer);
+  });
   return app;
 }
 
+export function idpMetadataUpdater(
+  app: Express,
+  listener?: () => void
+): NodeJS.Timer {
+  return setInterval(async () => {
+    log.info("Restart spid strategy initialization ...");
+
+    container.cache.delete(SPID_STRATEGY);
+
+    const strategy = await container.resolve<Promise<passport.Strategy>>(
+      SPID_STRATEGY
+    );
+    passport.unuse("spid");
+    passport.use("spid", strategy);
+
+    const spidAuth = passport.authenticate("spid", { session: false });
+
+    // Override /login route with new updated spidAuth middleware
+    // tslint:disable: no-any
+    app._router.stack.reduce((prev: ReadonlyArray<any>, _: any) => {
+      if (_.route && _.route.path !== "/login") {
+        return [...prev, _];
+      }
+      return prev;
+    }, []);
+
+    app.get("/login", spidAuth);
+    log.info("Spid strategy initialization complete.");
+    if (listener) {
+      listener();
+    }
+  }, container.resolve<number>(IDP_REFRESH_TIME));
```

Comment:
I suggest to do the `container.resolve` outside `idpMetadataUpdater` and make `idpMetadataUpdater` accept a refresh time parameter

Reply:
fixed in 82e8752